### PR TITLE
Fix namespaces for files_external

### PR DESCRIPTION
**Pull request by RobinMcCorkell**
We had a bunch of CLASSPATH entries, for storage classes (and the Api connector, for some reason). I've moved all the storage classes to a subdirectory of lib/ to tidy things up, and corrected the namespaces as necessary so that the autoloader just works.

This is possible now (as opposed to before the backend registration overhaul) since the identifier saved in the config is separate from the actual class used. Thanks to identifier aliases, even if someone has an old config containing one of these class names, it will be correctly detected.

cc @PVince81 @rullzer @icewind1991 @MorrisJobke 

Also cc @jmaciasportela in case WND needs a fixup (it probably has \OC\Files\Storage\SMB somewhere, just change that to \OCA\Files_External\Lib\Storage\SMB)

### CODE REVIEW

**Comment by rullzer** [comment]
Nice! Getting rid of all those CLASSPATh definitions is very welcome!

It all looks good to me (and CI seems to agree). But lets await the judge ments of the wielder of ownCloud storages @icewind1991

File: apps/files_external/appinfo/routes.php [hunk 1]
```diff
@@ -61,6 +61,6 @@
 
 \OCP\API::register('get',
 		'/apps/files_external/api/v1/mounts',
-		array('\OCA\Files\External\Api', 'getUserMounts'),
+		array('\OCA\Files_External\Lib\Api', 'getUserMounts'),
```
Comment (nickvergessen) [inline apps/files_external/appinfo/routes.php]: Btw you don't need Lib in the namespace, we find it without `Lib` in the lib folder as well.
https://github.com/owncloud/core/blob/master/lib/autoloader.php#L125-L133

**Comment by nickvergessen** [comment]
Please also drop a mail to the mailing list, just in case someone is using private classes

File: apps/files_external/appinfo/routes.php (hunk 1 above)
Comment (Xenopathic) [inline apps/files_external/appinfo/routes.php]: All the other classes use Lib in the namespace, I was just being consistent :smile: (also I kinda like the explicitness)

**Comment by Xenopathic** [comment]
@nickvergessen Should I drop an email now, or wait until the needed 2 reviews come through and this PR is ready for merge?

**Comment by nickvergessen** [comment]
When it's merged

File: apps/files_external/appinfo/routes.php (hunk 1 above)
Comment (MorrisJobke) [inline apps/files_external/appinfo/routes.php]: I also makes then the autoloader move to PSR-4 easier ;)

Comment (nickvergessen) [inline apps/files_external/appinfo/routes.php]: yeah fine by me, just wanted to clearify

**Comment by icewind1991** [comment]
Looks good :+1:

**Comment by rullzer** [comment]
:+1:

File: apps/files_external/appinfo/app.php [hunk 2]
```diff
@@ -26,17 +26,7 @@
...
-OC::$CLASSPATH['OC\Files\Storage\FTP'] = 'files_external/lib/ftp.php';
-OC::$CLASSPATH['OC\Files\Storage\OwnCloud'] = 'files_external/lib/owncloud.php';
-OC::$CLASSPATH['OC\Files\Storage\Google'] = 'files_external/lib/google.php';
-OC::$CLASSPATH['OC\Files\Storage\Swift'] = 'files_external/lib/swift.php';
-OC::$CLASSPATH['OC\Files\Storage\SMB'] = 'files_external/lib/smb.php';
-OC::$CLASSPATH['OC\Files\Storage\AmazonS3'] = 'files_external/lib/amazons3.php';
-OC::$CLASSPATH['OC\Files\Storage\Dropbox'] = 'files_external/lib/dropbox.php';
-OC::$CLASSPATH['OC\Files\Storage\SFTP'] = 'files_external/lib/sftp.php';
 OC::$CLASSPATH['OC_Mount_Config'] = 'files_external/lib/config.php';
-OC::$CLASSPATH['OCA\Files\External\Api'] = 'files_external/lib/api.php';
```
Comment (DeepDiver1975) [inline apps/files_external/appinfo/app.php]: These class names used to be part of the config if I remember correctly - is this still the case?

@icewind1991 @Xenopathic please clarify

**Comment by DeepDiver1975** [comment]
Please only merge after my question was answered - THX

**Comment by jmaciasportela** [comment]
cc @blizzz FYI

File: apps/files_external/appinfo/app.php (hunk 2 above)
Comment (blizzz) [inline apps/files_external/appinfo/app.php]: I checked config in master, 9.0 and 8.2 and cannot see traces of the class name in the config tables or files.

With a simple SMB mount i could switch the branch and it continued to work fine. :+1:

Comment (Xenopathic) [inline apps/files_external/appinfo/app.php]: The class names were used up until the backend registration refactor, after which the config used identifiers which are defined by the backend registration classes (lib/backend/). That's what allows us to decouple the class names from how the config stores them. Also, in case we are upgrading from a previous version in which class names are actually used, we still register identifier aliases to the old class names so they get resolved properly.

TL;DR: It's all fine :smile:

**Comment by MorrisJobke** [comment]
> Please only merge after my question was answered - THX

Question answered -> merg

File: apps/files_external/appinfo/app.php (hunk 2 above)
Comment (DeepDiver1975) [inline apps/files_external/appinfo/app.php]: Nice job! Thanks!

**Comment by LukasReschke** [comment]
This broke Google Drive: https://github.com/owncloud/core/issues/24407
